lib/coverband.rb

```diff
@@ -66,10 +66,16 @@ def self.configuration
 
   def self.start
     Coverband::Collectors::Coverage.instance
-    # TODO: Railtie sets up at_exit after forks, via middleware, perhaps this hsould be
+    # TODO: Railtie sets up at_exit after forks, via middleware, perhaps this should be
     # added if not rails or if rails but not rackserverrunning
-    AtExit.register
-    Background.start if configuration.background_reporting_enabled && !RackServerCheck.running?
+    AtExit.register unless tasks_to_ignore?
+    Background.start if configuration.background_reporting_enabled && !RackServerCheck.running? && !tasks_to_ignore?
+  end
+
+  def self.tasks_to_ignore?
+    (defined?(Rake) &&
+    Rake.respond_to?(:application) &&
```

Comment:
Don't think we need this check. Seems like it's always had an application. This commit moves the method to a different file in 2009 but it was already there at that point:
https://github.com/ruby/rake/commit/2de58bc9e1b1ab89564f160c58f768cefa736e70